mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp

```diff
@@ -375,15 +375,9 @@ LogicalResult WarpgroupGenerateDescriptorOp::verify() {
   MemRefType memrefType = getTensor().getType();
   MemRefType tensorMapType = getTensorMap().getType().getTensor();
 
-  if (memrefType != tensorMapType)
-    return emitError() << "memref and tensor map type mismatch";
-
   if (!memrefType.hasStaticShape() || !tensorMapType.hasStaticShape())
     return emitError() << "supports only static shapes";
 
-  if (memrefType.getRank() != 2)
-    return emitError() << "supports only 2d memref is supported for now";
```

Comment:
Since this is supposed to be used to feed into wgmma operations, why do we need to support more than 2ds?

(Sorry for the dumb questions x).)